erpnext/crm/doctype/opportunity/opportunity.py

```diff
@@ -565,6 +602,13 @@ def submit_communication(name, contact_date, remarks, submit_follow_up=False):
 
 	comm.insert(ignore_permissions=True)
 
+	if cint(submit_follow_up):
+		follow_up = [f for f in opp.contact_schedule if not f.contact_date]
+		if follow_up:
+			follow_up[0].contact_date = getdate(contact_date)
+
+	opp.save()
```

Comment:
Better keep this inside conditional block .. since we only need to save the opportunity if we're changing something